### DISCUSSION TITLE
[Bug]:   Emulator unable to read save data for Pokemon Crystal after opening a 2nd instance of Visual Boy Advance-M

### DISCUSSION THREAD

**Discussion by Garblits** [visualboyadvance-m/visualboyadvance-m#1176] [General]
### Expected behavior?

I expected to be able to open two instances of VBA-M to try trading pokemon between Pokemon Crystal, and Pokemon Red, and to be able to use the save data for both ROM's.

### Actual behavior?

After opening two instances of VBA-M, and restarting the games, the save data was no longer able to be read, despite the file still being in the folder. I was able to recover the save data for Crystal by opening a previous save state and manually saving, but the save file for Red was unable to be recovered.

### Steps to reproduce the problem

1.Open 2 instances of VBA-M without either ROM loaded.
2.Open ROM for Pokemon Crystal
3.Upon opening the ROM, the save data is no longer able to be read. Save states could be loaded however.
4.
5.
...

### Error messages?

If this is a crash bug, and you have the time and inclination, please provide a symbolic backtrace.
Instructions for doing so are [here](https://github.com/visualboyadvance-m/visualboyadvance-m/blob/master/README.md#reporting-crash-bugs).

### What game were you playing

<Some of this information can be provided with a screenshot of Rom Information, SAVE TYPE for GBA titles can be obtained from Options > Game Boy Advance > Config>
Title: Pokemon Crystal
GB, GBC or GBA: GBC
ROM CODE: 
BIOS Used?: (Share crc hash)
SAVE TYPE: (Attach Battery or Savestate here)
Clean, Intro'd or Hacked?:(Attach Patch here)

### What settings are you using?

Video driver (simple, opengl): opengl
Sound driver (directsound, xaudio, openal, SDL): SDL
Scale Filter Used?:

### Specifications and Build details

VBA-M: (include commit version number and whether you are using the 32-bit or 64-bit version)
OS: (include version and OS build number using WinVer if on Windows):
If using Linux, specify if using xorg or Wayland:
CPU:
GPU:
RAM:

### Copyright Compliance

- [X] There are no infringing files attached to this bug report.

**Comment by Squall-Leonhart** [comment]
The second instance will accept a save file with the SA# extension where # is a number corresponding to the number of instances of the emulator.

When restarted with Linking enabled in a single instance, the save file is always SA1 instead of SAV.

**Reply by Squall-Leonhart** [reply]
https://github.com/visualboyadvance-m/visualboyadvance-m/issues/867 is open for the issue where the Import feature does not write the SA# save while linking is active.

**Comment by Squall-Leonhart** [comment]
That said, GB linking doesn't work anyway at the current time and there is nobody dedicating time to looking at it either.

**Comment by Garblits** [comment]
Well thanks for the help but it's also seemed to have wiped all the pokemon from my box as well. I think I'm just gonna start fresh using a different emulator.
